Sources/ComposableArchitecture/Store.swift

```diff
@@ -234,66 +264,95 @@ public final class Store<State, Action> {
   ) -> Effect<Store<LocalState, Action>, Never> {
     self.producerScope(state: toLocalState, action: { $0 })
   }
-
+  
   func send(_ action: Action) {
     if !self.isSending {
       self.synchronousActionsToSend.append(action)
     } else {
       self.bufferedActions.append(action)
       return
     }
-
+    
     while !self.synchronousActionsToSend.isEmpty || !self.bufferedActions.isEmpty {
       let action =
         !self.synchronousActionsToSend.isEmpty
         ? self.synchronousActionsToSend.removeFirst()
         : self.bufferedActions.removeFirst()
-
+      
       self.isSending = true
       let effect = self.reducer(&self.state, action)
       self.isSending = false
-
+      
+      var didComplete = false
+      let effectID = UUID()
+      
       var isProcessingEffects = true
-      effect.startWithValues { [weak self] action in
-        if isProcessingEffects {
-          self?.synchronousActionsToSend.append(action)
-        } else {
-          self?.send(action)
+      
+      let observer = Signal<Action, Never>.Observer(
+        value: { [weak self] action in
+          if isProcessingEffects {
+            self?.synchronousActionsToSend.append(action)
+          } else {
+            self?.send(action)
+          }
+        },
+        failed: .none,
+        completed: { [weak self] in
+          didComplete = true
+          self?.effectDisposables.removeValue(forKey: effectID)?.dispose()
+        },
+        interrupted: { [weak self] in
+          didComplete = true
+          self?.effectDisposables.removeValue(forKey: effectID)?.dispose()
         }
```

Comment:
In this case we do need the full `completed:` and `interrupted:` handling since the logic is more than just disposing the subscription.